migrate/live-migration/live-migration-from-postgres.md

```diff
@@ -0,0 +1,252 @@
+---
+title: Migrate from PostgreSQL using live migration
+excerpt: Migrate from a PostgreSQL database using the low-downtime live migration method
+products: [cloud]
+keywords: [migration, low-downtime]
+tags: [migration, logical backup, replication]
+---
+
+import GettingHelp from "versionContent/_partials/_migrate_dual_write_backfill_getting_help.mdx";
+import SourceTargetNote from "versionContent/_partials/_migrate_source_target_note.mdx";
+import StepOne from "versionContent/_partials/_migrate_dual_write_step1.mdx";
+
+# Live migration from PostgreSQL database with pgcopydb
+
+This document provides detailed step-by-step instructions to migrate data using
+[pgcopydb][pgcopydb] to perform a live migration from a source database which
+is using PostgreSQL to Timescale.
+
+Before beginning the migration process, ensure that the following tools are
+installed and available on the system that performs the migration:
+
+- `psql`, `pg_dump`, and `pg_dumpall`: These can be installed by following the
+  instructions in the [How to Install psql on Mac, Ubuntu, Debian, Windows][install-psql]
+  blog post. Although the instructions specifically mention `psql`, following
+  them also installs `pg_dump` and `pg_dumpall`.
+
+- `pgcopydb`: Installation instructions can be found in the
+  [official repository][install-pgcopydb].
+
+<SourceTargetNote />
+
+<!-- TODO:  what about schema changes are they allowed? What other conditions
+are there? -->
+
+In detail, the migration process consists of the following steps:
+
+1. Set up a target database instance in Timescale.
+1. Set up a replication slot and snapshot.
+1. Migrate roles and schema from source to target.
+1. Convert hypertables and enable Timescale features.
+1. Migrate initial data from source to target.
+1. Apply replication changes.
+1. Promote target database as new primary.
+
+<GettingHelp />
+
+<StepOne />
+
+## 2. Set up a replication slot and snapshot
+
+Use the `pgcopydb follow` to create a [replication slot][replication-slot] and
+a [snapshot][snapshot].
+
+```sh
+pgcopydb follow \
+  --source "$SOURCE" \
+  --target "$TARGET" \
+  --fail-fast \
+  --plugin wal2json
```

Comment:
(Questios for my own interest): Is wal2json available in all Postgres versions? What is the minimum Postgres version that we need for live migration?

Reply:
I just did a quick read and it's an external extension, I think it's installed with the postgres-dev package. From the pgcopydb's docs I get that we can use `test_decoding` and that one I think comes bundle with postgres and it's the default for pgcopydb.

@arajkumar is there a reason why we are using wal2json and not the other? WDYT

Reply:
There was a bug in pgcopydb while using with test_decoding which prevents migration of a table which has reserved Postgres keywords as column name.(e.g. time). I [fixed it in upstream](https://github.com/dimitri/pgcopydb/pull/482), but it is not yet released to the public.

We have to stick to wal2json until  the new pgcopydb release is made available.

wal2json is usually distributed with most of the postgres distributions. Most of the cloud providers(AWS, GCP, Azure) has enabled wal2json in their managed offerings.